### DISCUSSION TITLE
Second Order Lagrange Divergence

### DISCUSSION THREAD

**Discussion by jinca** [idaholab/moose#20808] [Q&A Modules: General] [answered]
Hello dear MOOSE team
I have a question about the convergence status in a simulation that calculates the temperature and displacement in 3D.
- When I am using 2 cores and the FIRST LAGRANGE ORDER, it converged! 
- When I am using 32 cores and the SECOND LAGRANGE ORDER, it does not!  
I am attaching the output file for both cases. 
I cannot provide the mesh by applying the second lagrange order because the simulation was stopped at 12 hours. 
(I set my script for 12 hours)
Thanks in advance,
Julita
[hypre_2_FIRST.txt](https://github.com/idaholab/moose/files/8505275/hypre_2_FIRST.txt)
[hypre_32_SECOND.txt](https://github.com/idaholab/moose/files/8505276/hypre_32_SECOND.txt)

**Comment by GiudGiud** [comment, marked as answer]
hello

The residual keeps decreasing, it is converging. You need to play with the convergence criteria to give it more chance to converge (increase l_max_its and nl_max_its) or to accept a less-converged solution (increase l_tol and nl_abs/rel_tol)

See this page for all the parameters.
https://mooseframework.inl.gov/source/executioners/Steady.html

Guillaume
